File: modulo_6/mario.py
def main():
    # Etapa 1: Solicitar altura do usuário
    height = get_height()

    # Etapa 2: Construir a pirâmide dupla
    for row in range(1, height + 1):
        # Calcular espaços antes da primeira pirâmide
        spaces = " " * (height - row)
        # Gerar os hashes para a primeira pirâmide
        left_hashes = "#" * row
        # Espaços fixos entre as pirâmides
        gap = "  "
        # Gerar os hashes para a segunda pirâmide
        right_hashes = "#" * row
        # Concatenar e imprimir a linha
        print(spaces + left_hashes + gap + right_hashes)


def get_height():
    """Solicita ao usuário uma altura válida entre 1 e 8"""
    while True:
        try:
            # Solicitar entrada
            height = int(input("Height: "))
            # Verificar se está no intervalo permitido
            if 1 <= height <= 8:
                return height
        except ValueError:
            pass  # Ignorar entradas não inteiras
        # Exibir mensagem de erro e repetir solicitação
        print("Invalid input. Please enter a number between 1 and 8.")


# Executar o programa principal
main()

File: modulo_1-C/mario.c
#include <cs50.h>
#include <stdio.h>

int main(void)
{

    int height;

    // Solicita ao usuário o tamanho da pirâmide entre 1 e 8
    do
    {
        height = get_int("Tamanho da Pirâmide: ");
    }
    while (height < 1 || height > 8);

    // Construção da piramide
    for (int i = 1; i <= height; i++)
    {

        // Dando os espaços a esquerda para a construção da pirâmide
        for (int j = 0; j < height - i; j++)
        {
            printf(" ");
        }

        // Imprime os # para a construção da pirâmide
        for (int k = 0; k < i; k++)
        {
            printf("#");
        }

        // Quebra de linha para a proxima linha da pirâmide
        printf("\n");
    }
}
